cmd/opm/index/add.go

```diff
@@ -226,3 +255,61 @@ func getContainerTools(cmd *cobra.Command) (string, string, error) {
 
 	return pullTool, buildTool, nil
 }
+
+func validateBundles(bundles []string, skipTLS bool, rootCA string) error {
+	logger := logrus.WithFields(logrus.Fields{"bundles": bundles})
+	ctx := context.Background()
+	registryOpts := []containerdregistry.RegistryOption{containerdregistry.SkipTLS(skipTLS)}
+	if !skipTLS && rootCA != "" {
+		rootCAs := x509.NewCertPool()
+		certs, err := ioutil.ReadFile(rootCA)
+		if err != nil {
+			return err
+		}
+		if !rootCAs.AppendCertsFromPEM(certs) {
+			return fmt.Errorf("failed to fetch root CA from %s", rootCA)
+		}
+		registryOpts = append(registryOpts, containerdregistry.WithRootCAs(rootCAs))
+	}
+	registry, err := containerdregistry.NewRegistry(registryOpts...)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := registry.Destroy(); err != nil {
+			logger.Error(err)
+		}
+	}()
+	var errors []error
+	for _, b := range bundles {
+		bundleRef := image.SimpleReference(b)
+		if err := registry.Pull(ctx, bundleRef); err != nil {
```

Comment:
This feels unnecessary? We already pull the bundle later in the add command and parse the manifests. Shouldn't we just have to do that once?